src/lib.rs

```diff
@@ -160,12 +158,16 @@ impl Tracer {
         if self.tracer_ctx.is_some() {
             return Err(TraceMeError::TracerAlreadyStarted);
         }
+        if !self.trace_filename.ends_with(".ptt") {
```

Comment:
I wonder if you really want to use `Path::extension` on this? The nature of filename extensions are OS dependant...